src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -35,7 +36,7 @@ private RazorLanguageServer()
         {
         }
 
-        public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Trace trace)
+        public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Trace trace, Version version = null)
```

Comment:
I'd rename `version` to `defaultProtocolVersion` since it represents a part of the protocol that we fall back to when `ClientInfo` isn't provided.

I thought a bit more about this too and another route here would be to update VS' language server protocol extension to pass this along. I wonder how hard that'd be.